.circleci/config.yml

```diff
@@ -1345,6 +1345,41 @@ jobs:
           name: Verify Values Match
           command: |
             ./ops/scripts/ci-match-values-between-files.sh "<< parameters.file1_path >>" "<< parameters.pattern_file1 >>" "<< parameters.file2_path >>" "<< parameters.pattern_file2 >>"
+
+  test-proofs:
+    machine:
+      image: ubuntu-2204:2023.07.2
+    steps:
+      - checkout
+      - check-changed:
+          patterns: contracts-bedrock/test/kontrol,contracts-bedrock/src/L1/OptimismPortal\.sol
+      - run:
+          name: "Start Docker Container"
+          command: |
+            export KONTROL_VERSION=$(cat ../../.kontrolrc)
+            docker run --name optimism-ci \
+                    --rm \
+                    -v $(pwd)/kout/proofs:/home/user/workspace/packages/contracts-bedrock/kout/proofs/ \
+                    --interactive \
+                    --tty \
+                    --detach \
+                    --user root \
+                    --workdir /home/user/workspace \
+                    runtimeverificationinc/kontrol:ubuntu-jammy-${KONTROL_VERSION}
+            docker cp . optimism-ci:/home/user/workspace
+            docker exec optimism-ci chown -R user:user /home/user
+      - run:
+          name: 'Run Kontrol'
+          command: |
+            docker exec -u user optimism-ci bash -c 'cd packages/contracts-bedrock && ./test/kontrol/kontrol/run-kontrol.sh'
+      - run:
+          name: Create Results Tar
+          command: |
+            tar -czvf results.tar.gz kout/proofs
+      - store_artifacts:
+          path: results.tar.gz
+          destination: results
```

Comment:
Yea that's accurate. But I think always uploading is also ok, because maybe a proof passed unexpectedly that we want to debug. My suggestion would be to leave it as "always upload" which should be simpler, and we can improve this in the future if needed